### PR TITLE
MON-3915: chore: bump Go dependencies

### PR DESCRIPTION
**Pull request by simonpasquier**
* [ ] I added CHANGELOG entry for this change.
* [x] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -9,34 +11,34 @@ require (
...
-	golang.org/x/text v0.14.0
+	golang.org/x/text v0.15.0
 	gopkg.in/yaml.v2 v2.4.0
 	gopkg.in/yaml.v3 v3.0.1
-	k8s.io/api v0.29.0
-	k8s.io/apiextensions-apiserver v0.29.0
-	k8s.io/apimachinery v0.29.0
-	k8s.io/apiserver v0.29.0
+	k8s.io/api v0.30.2
+	k8s.io/apiextensions-apiserver v0.30.1
```
Comment (rexagod) [inline go.mod:32]: (nit) There seems to be a newer `v0.30.2` for [`k8s.io/apiextensions-apiserver`](https://pkg.go.dev/k8s.io/apiextensions-apiserver).

File: go.mod [hunk 2]
```diff
@@ -9,34 +11,34 @@ require (
...
+	k8s.io/apiserver v0.30.2
 	k8s.io/autoscaler/vertical-pod-autoscaler v1.0.0
-	k8s.io/client-go v0.29.0
-	k8s.io/component-base v0.29.0
-	k8s.io/klog/v2 v2.110.1
-	k8s.io/kube-aggregator v0.29.0
+	k8s.io/client-go v0.30.2
+	k8s.io/component-base v0.30.2
+	k8s.io/klog/v2 v2.120.1
+	k8s.io/kube-aggregator v0.30.1
```
Comment (rexagod) [inline go.mod:39]: (nit) There seems to be a newer `v0.30.2` for [`k8s.io/kube-aggregator`](https://pkg.go.dev/k8s.io/kube-aggregator).

**Review by rexagod — APPROVED** [review]

**Review by marioferh — COMMENTED** [review]
/lgtm

File: go.mod [hunk 3]
```diff
@@ -1,6 +1,8 @@
 module github.com/openshift/cluster-monitoring-operator
 
-go 1.22
+go 1.22.0
+
+toolchain go1.22.4
```
Comment (jan--f) [inline go.mod:5]: Will we need to adjust any of our PR checks to guard against accidentally bumping the toolchain directive?
Or is there anything to consider how ART deals with this?

**Pull request merged**